test/functional/mempool_updatefromblock.py

```diff
@@ -36,87 +34,62 @@ def transaction_graph_test(self, size, n_tx_to_mine=None, start_input_txid='', e
+        self.wallet.scan_blocks(start=76, num=1)
+        node.generate(size)
+        txs = []
+        block_ids = []
+        tx_size = []
+        for i in range(size):
+            tx = self.wallet.send_self_transfer(from_node=node)
+            m_tx = node.getrawmempool(True)[tx['txid']]
+            txs.append(tx)
+            tx_size.append(m_tx['vsize'])
+            if (i+1) % n_tx_to_mine == 0:
+                assert_equal(len(node.getrawmempool()), n_tx_to_mine)
+                self.log.info('The batch of {} transactions has been accepted into the mempool.'
+                              .format(n_tx_to_mine))
+                block_ids.append(self.wallet.generate(1)[0])
+                assert_equal(len(node.getrawmempool()), 0)
+                self.log.info(
+                    'All of the transactions from the current batch have been mined into a block.')
+
+        self.log.info(
+            'Mempool size pre-invalidation: {}'.format(len(node.getrawmempool())))
+        assert_equal(len(node.getrawmempool()), 0)
+        # Invalidate the first block to send the transactions back to the mempool.
+        start = time.time()
+        node.invalidateblock(block_ids[0])
+        end = time.time()
+        self.log.info(
+            'Mempool size post-invalidation: {}'.format(len(node.getrawmempool())))
+        assert_equal(len(node.getrawmempool()), size)
+        self.log.info('All of the recently mined transactions have been re-added into the mempool in {} seconds.'
+                      .format(end - start))
+
+        self.log.info(
+            'Checking descendants/ancestors properties of all of the in-mempool transactions...')
+        for k, tx in enumerate(txs):
+            id = tx['txid']
             self.log.debug('Check transaction #{}.'.format(k))
-            assert_equal(self.nodes[0].getrawmempool(True)[tx]['descendantcount'], size - k)
-            assert_equal(self.nodes[0].getrawmempool(True)[tx]['descendantsize'], sum(tx_size[k:size]))
-            assert_equal(self.nodes[0].getrawmempool(True)[tx]['ancestorcount'], k + 1)
-            assert_equal(self.nodes[0].getrawmempool(True)[tx]['ancestorsize'], sum(tx_size[0:(k + 1)]))
+            assert_equal(node.getrawmempool(True)[id]['descendantcount'], size - k)
+            assert_equal(node.getrawmempool(True)[id]['descendantsize'], sum(tx_size[k:size]))
+            assert_equal(node.getrawmempool(True)[id]['ancestorcount'], k + 1)
+            assert_equal(node.getrawmempool(True)[id]['ancestorsize'], sum(tx_size[0:(k + 1)]))
 
     def run_test(self):
-        # Use batch size limited by DEFAULT_ANCESTOR_LIMIT = 25 to not fire "too many unconfirmed parents" error.
-        self.transaction_graph_test(size=100, n_tx_to_mine=[25, 50, 75])
+        # Use batch size limited by DEFAULT_ANCESTOR_LIMIT = 25
+        # to not fire "too many unconfirmed parents" error.
+        self.transaction_graph_test(size=100, n_tx_to_mine=25)
```

Comment:
why is this changed?

Reply:
The goal should probably be to fix the error, not to remove iterations from the test. I guess you can create more outputs or mine a block or something?

Reply:
Ah ok, apologies for the noise. Will put this into draft mode since some work is still needed. Thanks for review!